set_bucket_tags/app.py

```diff
@@ -54,27 +53,49 @@ def get_principal_id(tags):
     raise ValueError('Could not derive a provisioningPrincipalArn from tags')
 
 
-def get_synapse_email(synapse_id):
-  '''Derive an email from a id'''
+def get_synapse_userProfile(synapse_id):
+  '''Get synapse user profile data'''
   syn = synapseclient.Synapse()
   user_profile = syn.getUserProfile(synapse_id)
   log.debug(f'Synapse user profile: {user_profile}')
-  user_name = user_profile.get('userName')
-  synapse_email = f'{user_name}@synapse.org'
-  return synapse_email
 
+  return user_profile
 
-def add_owner_email_tag(tags, email):
-  '''Add an OwnerEmail tag to set of bucket tags'''
-  owner_email_tag = next((tag for tag in tags if tag['Key'] == 'OwnerEmail'), None)
-  if owner_email_tag:
-    owner_email_tag['Value'] = email
-  else:
-    new_owner_email_tag = { 'Key': 'OwnerEmail', 'Value': email}
-    tags.append(new_owner_email_tag)
+
+def get_synapse_tags(user_profile):
+  '''Derive synapse tags from synapse user profile data'''
+
+  tags = []
+  IGNORE_KEYS = ["createdOn"]
+  for key, value in user_profile.items():
+    if key in IGNORE_KEYS:
+      continue
+
+    # derive synapse email tag based on userName
+    if key == "userName":
+      synapse_email = f'{value}@synapse.org'
+      tags.append({'Key': 'synapse:email', 'Value': synapse_email})
+      tags.append({'Key': 'OwnerEmail', 'Value': synapse_email})  # legacy
+
+    tag = {'Key': f'synapse:{key}', 'Value': value}
+    tags.append(tag)
+
+  log.debug(f'Synapse tags: {tags}')
   return tags
 
 
+def merge(list_x, list_y):
+  '''Merge two list of tags'''
+  z = []
+  for x in list_x:
+    z.append(x)
+
+  for y in list_y:
+    z.append(y)
+
+  return z
```

Comment:
you can shorten this merge function to `list_x + list_y`

Reply:
thanks.  fixed.